doc/zmq_getsockopt.txt

```diff
@@ -983,6 +983,19 @@ Default value:: 8192
 Applicable socket types:: All, when using TCP, IPC, PGM or NORM transport.
 
 
+ZMQ_SUBSCRIPTION_COUNT: Number of subscriptions received
```

Comment:
"subscription count" is ambiguous as it might be thought to refer to the number of subscribers, but this is just counting the number of different topics

Reply:
however the way to subscribe is to use zmq_setsockopt(... ZMQ_SUBSCRIBE ... )... so using "subscription count" as term to refer to the number of different ZMQ_SUBSCRIBE operations (*) was intuitive at least for me :)

(*) I'm assuming it's not a common habit to subscribe multiple times to the same topic prefix... IOW I'm assuming that most likely the number of times zmq_setsockopt(... ZMQ_SUBSCRIBE ... ) is invoked on the SUB socket will match the result of zmq_getsockopt(.. ZMQ_SUBSCRIPTION_COUNT.. ) on the PUB socket (if network works fine)

Anyway I'm open to any other suggestion for the name... ZMQ_PUBSUB_TOPIC_PREFIXES ? ZMQ_TOPIC_COUNT?

Reply:
ZMQ_PUB_TOPICS_COUNT?

Reply:
fine for me, will update this PR accordingly!

Reply:
I guess that if we call this socket option ZMQ_PUB_* we don't want to support this socket option also on SUB sockets right?

Reply:
hi @bluca ,
can you review again?
 
latest changes include:
 * renamed ZMQ_SUBSCRIPTION_COUNT -> ZMQ_TOPICS_COUNT
 * implemented ZMQ_TOPICS_COUNT for (X)SUB sockets by making a) atomic the size() method of the radix_tree_t and b) creating a num_prefixes() atomic method for trie_t, obtained by wrapping the generic trie-node trie_t inside a trie_with_size_t class
  * inside new test, invoke ZMQ_TOPICS_COUNT on the subscriber, not just on the publisher

I just wonder why the XSUB socket is using the trie_t when there is also mtrie_t,,,